--- a/src/main.c
+++ b/src/main.c
@@ -805,11 +805,7 @@ main(int argc, char **argv)
 
 #if ENABLE_MODULE_IF
     cf.stable->modules_cf = rtpp_module_if_ctor(cf.stable, cf.stable->glog,
-#if !RTPP_DEBUG
-      "../modules/acct_csv/.libs/rtpp_acct_csv.so");
-#else
-      "../modules/acct_csv/.libs/rtpp_acct_csv_debug.so");
-#endif
+    "/usr/lib/x86_64-linux-gnu/rtpproxy/rtpp_acct_csv.so");
 #endif
 
     cf.stable->rtpp_cmd_cf = rtpp_command_async_ctor(&cf);
